src/Microsoft.AspNetCore.Session/CookieProtection.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Text;
+using Microsoft.AspNetCore.DataProtection;
+
+namespace Microsoft.AspNetCore.Session
+{
+    internal static class CookieProtection
+    {
+        internal static string Protect(IDataProtector protector, string data)
+        {
+            if (protector == null)
+            {
+                throw new ArgumentNullException(nameof(protector));
+            }
+            if (string.IsNullOrEmpty(data))
+            {
+                return data;
+            }
+
+            var userData = Encoding.UTF8.GetBytes(data);
+
+            var protectedData = protector.Protect(userData);
+            return Convert.ToBase64String(protectedData).TrimEnd('=').Replace('+', '-').Replace('/', '_');
+        }
+
+        internal static string Unprotect(IDataProtector protector, string protectedText)
+        {
+            try
+            {
+                if (string.IsNullOrEmpty(protectedText))
+                {
+                    return string.Empty;
+                }
+
+                var protectedData = Convert.FromBase64String(Pad(protectedText.Replace('-', '+').Replace('_', '/')));
+                if (protectedData == null)
+                {
+                    return string.Empty;
+                }
+
+                var userData = protector.Unprotect(protectedData);
+                if (userData == null)
+                {
+                    return string.Empty;
+                }
+
+                return Encoding.UTF8.GetString(userData);
+            }
+            catch
+            {
+                // TODO trace exception, but do not leak other information
```

Comment:
Should we add the tracing now? Certainly would be useful if folks report problems in RC2...